lib/jekyll/commands/build.rb

```diff
@@ -72,7 +72,8 @@ def build(site, options)
         # Returns nothing.
         def watch(_site, options)
           External.require_with_graceful_fail "jekyll-watch"
-          Jekyll::Watcher.watch(options)
+          (watch = Jekyll::Watcher.method(:watch)).parameters.size == 1 ?
+              watch.call(options) : watch.call(options, _site)
```

Comment:
Hey @mojavelinux! We get a lot of Ruby newcomers and this syntax, while nicely concise, might be unwelcoming. Would you consider refactoring it? Very minor refactor, something like:

```ruby
watch = Jekyll::Watcher.method(:watch)
if watch.parameters.size == 1
  watch.call(options)
else
  watch.call(options, site)
end
```

We should also move `_site` to `site` now that we're using the argument (a Rubocop thing).